python/pluscodes.py

```diff
@@ -0,0 +1,515 @@
+    return code
+
+"""
+ Decodes an Open Location Code into the location coordinates.
+ Returns a CodeArea object that includes the coordinates of the bounding
+ box - the lower left, center and upper right.
+ Args:
+   code: The Open Location Code to decode.
+ Returns:
+   A CodeArea object that provides the latitude and longitude of two of the
+   corners of the area, the center, and the length of the original code.
+"""
+def decode(code):
+    if not isFull(code):
+        raise ValueError('Passed Open Location Code is not a valid full code - ' + str(code))
+    # Strip out separator character (we've already established the code is
+    # valid so the maximum is one), padding characters and convert to upper
+    # case.
+    code = re.sub('[+0]','',code)
+    code = code.upper()
+    # Decode the lat/lng pair component.
+    codeArea = decodePairs(code[0:PAIR_CODE_LENGTH_])
+    if len(code) <= PAIR_CODE_LENGTH_:
+        return codeArea
+     # If there is a grid refinement component, decode that.
+    gridArea = decodeGrid(code[PAIR_CODE_LENGTH_:])
+    return CodeArea(codeArea.latitudeLo + gridArea.latitudeLo,
+            codeArea.longitudeLo + gridArea.longitudeLo,
+            codeArea.latitudeLo + gridArea.latitudeHi,
+            codeArea.longitudeLo + gridArea.longitudeHi,
+            codeArea.codeLength + gridArea.codeLength)
+
+"""
+ Recover the nearest matching code to a specified location.
+ Given a short Open Location Code of between four and seven characters,
+ this recovers the nearest matching full code to the specified location.
+ The number of characters that will be prepended to the short code, depends
+ on the length of the short code and whether it starts with the separator.
+ If it starts with the separator, four characters will be prepended. If it
+ does not, the characters that will be prepended to the short code, where S
+ is the supplied short code and R are the computed characters, are as
+ follows:
+ SSSS    -> RRRR.RRSSSS
+ SSSSS   -> RRRR.RRSSSSS
+ SSSSSS  -> RRRR.SSSSSS
+ SSSSSSS -> RRRR.SSSSSSS
+ Note that short codes with an odd number of characters will have their
+ last character decoded using the grid refinement algorithm.
+ Args:
+   shortCode: A valid short OLC character sequence.
+   referenceLatitude: The latitude (in signed decimal degrees) to use to
+       find the nearest matching full code.
+   referenceLongitude: The longitude (in signed decimal degrees) to use
+       to find the nearest matching full code.
+ Returns:
+   The nearest full Open Location Code to the reference location that matches
+   the short code. Note that the returned code may not have the same
+   computed characters as the reference location. This is because it returns
+   the nearest match, not necessarily the match within the same cell. If the
+   passed code was not a valid short code, but was a valid full code, it is
+   returned unchanged.
```

Comment:
Because of issue #37, could you update the comment here (see the JS file for the changes)?